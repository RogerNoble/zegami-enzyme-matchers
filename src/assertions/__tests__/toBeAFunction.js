const React = require('react');
const { shallow } = require('enzyme');

const toBeAFunction = require('../toBeAFunction');

const Fixture = () => { return false; };

describe('toBeAFunction', () => {
  const truthyResults = toBeAFunction(Fixture);
  const falsyResults = toBeAFunction(Fixture);

  it('returns the pass flag properly', () => {
    expect(truthyResults.pass).toBeTruthy();
    expect(falsyResults.pass).toBeFalsy();
  });

  it('returns the message with the proper text', () => {
    expect(truthyResults.message).toMatchSnapshot();
  });

  it('returns the negated message with the proper text', () => {
    expect(truthyResults.negatedMessage).toMatchSnapshot();
  });

  it('provides contextual information for the message', () => {
    expect(truthyResults.contextualInformation).toMatchSnapshot();
  });
});
