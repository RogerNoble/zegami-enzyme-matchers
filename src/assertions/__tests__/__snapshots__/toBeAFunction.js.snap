// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toBeAFunction provides contextual information for the message 1`] = `
Object {
  "expected": " [empty set]",
}
`;

exports[`toBeAFunction returns the message with the proper text 1`] = `"Expected ([empty set]) to be a \\"Function\\" but it wasn't."`;

exports[`toBeAFunction returns the negated message with the proper text 1`] = `"Expected ([empty set]) not to be a \\"Function\\" but it was"`;
